### PR TITLE
OONI Glossary

### PR DESCRIPTION
**Pull request by agrabeli**
Special thanks to Anatol for contributing to this glossary! :)

### CODE REVIEW

File: content/about/glossary.md [hunk 1]
```diff
@@ -0,0 +1,973 @@
...
+* [VPN](#vpn)
+
+### API
+
+An Application Programming Interface (API) is an interface that enables
+programmers to write integration code between different services.
+
+The [OONI API](https://api.ooni.io/) provides an interface that enables users to
+request specific types of OONI data and to download such data in a specific
+format (JSON).
```
Comment (hellais) [inline content/about/glossary.md]: We should mention in here that the OONI API is not suitable for batch usage and point people instead to the MetaDB.

File: content/about/glossary.md [hunk 2]
```diff
@@ -0,0 +1,973 @@
...
+Each AS is responsible for delivering IP packets to a set of IP addresses that
+it manages.
+
+### ASN
+
+An Autonomous System Number (ASN) is a unique identifier of an autonomous system
+(AS). This number allows its autonomous system to exchange routing information
+with other systems.
+
+Internet Service Providers (ISPs) must have an officially registered ASN.
```
Comment (hellais) [inline content/about/glossary.md]: I am not sure this is actually a legal requirement. It's probably desirable for them to have one, but AFAIK you can be a ISP without having an ASN. It's also crucial I think to say that ISPs can have also more than one ASN.

File: content/about/glossary.md [hunk 3]
```diff
@@ -0,0 +1,973 @@
+# OONI Glossary
+
+This glossary contains *brief explanations* for terms used in OONI apps,
+methodologies, and research reports.
+
+If you would like to see additional terms in this glossary, please [open a pull request](https://github.com/TheTorProject/ooni-web) or [send us an email](https://ooni.org/about/#contact).
```
Comment (hellais) [inline content/about/glossary.md]: The PR link needs updating the repo is not `ooni/ooni.org`.

File: content/about/glossary.md [hunk 4]
```diff
@@ -0,0 +1,973 @@
...
+
+A simple Google search of an ASN number will show you which ISP it corresponds
+to. For example, if you search "AS30722", you will see that it belongs to
+"Vodafone Italia".
+
+### BGP
+
+The Border Gateway Protocol (BGP) is a standardized exterior gateway protocol
+designed to exchange routing and reachability information among autonomous
+systems (AS) on the internet.
```
Comment (hellais) [inline content/about/glossary.md]: I think if we are going for a simple definition it's not really useful to drop things like "standardized exterior gateway protocol
designed to exchange routing and reachability information among autonomous
systems (AS) on the internet".

I would say something along the lines of:

The Border Gateway Protocol (BGP) is a protocol used by Autonomous Systems (AS) on the internet to exchange routing information. With this information they are able to discover which AS is responsible for which set of IP address and what is the path IP packets need to take in order to reach that particular AS.

File: content/about/glossary.md (hunk 2 above)
Comment (bassosimone) [reply content/about/glossary.md]: My understanding is that you need an ASN to use BGP. It may be that you can be an ISP without BGP, but I think it seldom happens in practice, so I guess we should say that an ISP has one or more ASN.

File: content/about/glossary.md [hunk 5]
```diff
@@ -0,0 +1,975 @@
...
+
+### Client
+
+In the world of computers, a client is a piece of software or hardware that
+accesses a service made available by a server.
+
+The [OONI Probe app](https://ooni.org/install/), for example, is a client and it
+communicates with OONI servers in order to submit testing results.
+
+### Cloud-fronting
```
Comment (hellais) [inline content/about/glossary.md]: I suggest renaming this to Domain Fronting. Cloud fronting is derived from the AWS product called "AWS Cloud Front", while Domain Fronting is the non vendor specific term.

File: content/about/glossary.md [hunk 6]
```diff
@@ -0,0 +1,975 @@
...
+
+Cloud-fronting (commonly referred to as "domain-fronting") is a technique that
+circumvents internet censorship by appearing to connect to the domain of a cloud
+provider  -- such as google.com or awsstatic.com (Amazon cloud infrastructure)
+-- while actually connecting to a blocked website.
+
+This technique, therefore, not only provides users access to blocked internet
+services, but it also hides the fact that they are accessing those services at
+all.
```
Comment (hellais) [inline content/about/glossary.md]: I would refrase this section as follows:

Domain fronting (also sometimes known as "cloud fronting") is a censorship circumvention technique which relies on using a popular domain name hosted on a big cloud service provider to "front" traffic towards a circumvention service hosted on the same cloud service.

For example I may setup on the Google cloud a service under the domain `circumvention-service.google-cloud.com`, but front it via `google.com`. This means that from the perspective of somebody observing network traffic, all that they see is an encrypted TLS connection towards `google.com`, yet inside of this encrypted connection the client then says "actually I want to access `circumvention-service.google-cloud.com`", circumventing any potential block.

This means that the censor would either have to block all of the google cloud (and therefore disrupt many more service too as a result) or they will have a hard time distinguishing requests towards `circumvention-service.google-cloud.com` from requests to `google.com`. This concept take the name of "collateral freedom".

File: content/about/glossary.md [hunk 7]
```diff
@@ -0,0 +1,975 @@
...
+responsible for aggregating network measurement data from OONI Probe users and
+analyzing the data to identify network anomalies around the world.
+
+### DNS
+
+DNS stands for "Domain Name System" and it maps domain names to IP addresses.
+
+A domain is a name that we attribute to websites (when we create them), so that
+we can more easily remember and access them. For example, twitter.com is the
+domain of the Twitter website (`https://twitter.com/`).
```
Comment (hellais) [inline content/about/glossary.md]: I would drop the `https://twitter.com` from the parens as that is not the website, but rather the URL.

File: content/about/glossary.md [hunk 8]
```diff
@@ -0,0 +1,975 @@
...
+address of a particular domain name and your DNS resolver (the server managing
+your DNS requests) is dishonest, intentionally providing you an incorrect
+answer.
+
+As a result, you receive a forged response, such as the following:
+
+* **NXDOMAIN:** A response indicating that the requested domain is either not
+registered or invalid.
+
+* **Block page:** A web page that informs you that the requested domain has
```
Comment (hellais) [inline content/about/glossary.md]: This should say "The IP Address of a blockpage"

File: content/about/glossary.md [hunk 9]
```diff
@@ -0,0 +1,975 @@
...
+deliberately been blocked.
+
+* **Error page:** A web page that informs you that the requested domain does not
```
Comment (hellais) [inline content/about/glossary.md]: This is the same as the two mentioned above from the perspective of DNS.

File: content/about/glossary.md [hunk 10]
```diff
@@ -0,0 +1,975 @@
...
+exist or has been misspelled.
+
+When Internet Service Providers (ISPs) receive government orders to block
+specific websites, they sometimes adopt this technique. In these cases, it may
+be possible to circumvent the censorship by merely changing your DNS resolver.
```
Comment (hellais) [inline content/about/glossary.md]: add "or using encrypted DNS"

File: content/about/glossary.md [hunk 11]
```diff
@@ -0,0 +1,975 @@
...
+
+When you perform a DNS lookup, you ask your DNS resolver (the server managing
+your DNS requests) to disclose which IP addresses correspond to specific domain
+names.
+
+To check whether websites are blocked by means of DNS, [OONI Probe](https://ooni.org/install/) starts off by performing a DNS lookup to see
+which IP addresses are mapped to tested domains by the user's DNS resolver. If
+the IP addresses do *not* match those allocated by another DNS resolver (known
+to not implement censorship), it's possible that censorship is implemented at
+the DNS level.
```
Comment (hellais) [inline content/about/glossary.md]: Our test is a bit more complex than this, maybe it would be useful to mention that this is a simplified version of how the web_connectivity test works and link to the test specification for more details?

File: content/about/glossary.md [hunk 12]
```diff
@@ -0,0 +1,975 @@
...
+A DNS query (otherwise known as a "DNS request") is a request for information
+sent from a user's computer to a DNS server.
+
+In most cases, a DNS request is sent to ask for the IP address associated with a
+domain name (such as twitter.com).
+
+### DNS resolver
+
+A DNS resolver is a server that stores a central database of DNS nameservers
+(which define the DNS providers of domains), operating like an "address book".
```
Comment (hellais) [inline content/about/glossary.md]: I would rephrase this definition as:

A DNS resolver is a server which maps domain names to IP address, operating like an "address book".

Saying that it's a central database is technically wrong.

Also drop database from below.

File: content/about/glossary.md [hunk 13]
```diff
@@ -0,0 +1,975 @@
...
+* **DNS spoofing:** Where your DNS request is intercepted and you receive the
+wrong IP address.
+
+DNS tampering is a common censorship technique adopted by Internet Service
+Providers (ISPs) around the world.
+
+The [OONI Probe apps](https://ooni.org/install/) measure the [DNS tampering of websites and apps](https://ooni.org/nettest/).
+
+### Domain
+
```
Comment (hellais) [inline content/about/glossary.md]: This should be "Domain name"

File: content/about/glossary.md [hunk 14]
```diff
@@ -0,0 +1,975 @@
...
+A domain is a name that we attribute to websites (when we create them), so that
+we can more easily remember and access them.
+
+For example, **twitter.com** is the domain of the Twitter website
+(`https://twitter.com/`).
+
+### DPI
+
+Deep packet inspection (DPI) is an advanced method of examining and managing
```
Comment (hellais) [inline content/about/glossary.md]: I would drop "advanced method", depending on the type of DPI it canbe advanced or not.

File: content/about/glossary.md [hunk 15]
```diff
@@ -0,0 +1,975 @@
...
+All websites include an HTTP (or HTTPS) prefix (such as `http://example.com/`) so
+that your computer (the client) can request and receive the content of a website
+(hosted on a server).
+
+The transmission of data over the HTTP protocol is **unencrypted**.
+
+### HTTPS
+
+Hypertext Transfer Protocol Secure (HTTPS) -- also known as HTTP over TLS, or
+HTTP over SSL --  is the **encrypted version of the HTTP protocol**.
```
Comment (hellais) [inline content/about/glossary.md]: I would say rather than:

> encrypted version of the HTTP protocol

I would go for

> the HTTP protocol over an encrypted channel

File: content/about/glossary.md [hunk 16]
```diff
@@ -0,0 +1,975 @@
...
+An HTTPS collector is a server that collects data over the encrypted HTTPS
+protocol.
+
+OONI utilizes HTTPS collectors in order to collect network measurements from
+OONI Probe users around the world.
+
+### HTTP header
+
+An HTTP header is the section of the request and response messages that defines
+the operating parameters of an HTTP transaction between a client and a server.
```
Comment (hellais) [inline content/about/glossary.md]: I would rephrase this as:

HTTP headers are used to transmit to the server or the client metadata about the request or response. Certain HTTP headers are standardised for transmitting common bits of information, such as the `User-Agent` header field, which is used to tell the server which browser is performing the request.

File: content/about/glossary.md [hunk 17]
```diff
@@ -0,0 +1,975 @@
...
+
+When measuring the blocking of websites, the OONI Probe [Web Connectivity test](https://ooni.org/nettest/web-connectivity/) compares the HTTP headers of
+the tested website from the network of the user with a non-censored network (as
+part of a larger testing methodology).
+
+### HTTP request
+
+In order to connect to a server hosting an internet service (such as a website
+or application), you need to request access to it through the HTTP protocol.
+This is what we call an HTTP request.
```
Comment (hellais) [inline content/about/glossary.md]: I would drop this whole section because it's inaccurate.

File: content/about/glossary.md [hunk 18]
```diff
@@ -0,0 +1,975 @@
...
+website it is hosting.
+
+In some cases though, Internet Service Providers (ISP) prevent users from
+accessing certain websites by blocking or interfering with the connection
+between them and the server. This is a form of HTTP blocking,
+[measured](https://ooni.org/nettest/web-connectivity/) by the [OONI Probe app](https://ooni.org/install/).
+
+### HTTP response
+
+Every time you connect to a website, your browser (the client) sends a request
```
Comment (hellais) [inline content/about/glossary.md]: I would replace "connect" with "visit", because connect has a specific meaning in the context of internet protocols.

File: content/about/glossary.md [hunk 19]
```diff
@@ -0,0 +1,975 @@
...
+("HTTP request") through the HTTP protocol to the server which is hosting that
+website.
+
+A server normally responds with the content of the website it is hosting. This
```
Comment (hellais) [inline content/about/glossary.md]: I would rephrase this as:

A server will send in response to a HTTP requests a HTTP response, which includes HTTP response headers and optionally a response body, which in the case of a website will be the content of the page.

File: content/about/glossary.md [hunk 20]
```diff
@@ -0,0 +1,975 @@
...
+
+* HTTP Error *404*(Not Found): When you try to access a resource on a web
+server (usually a web page) that doesn’t exist (because, for example, you have
+mistyped a URL).
+
+* HTTP Error *400*(Bad Request): When the application you are using (such as
+your web browser) accesses a server incorrectly or the request was corrupted.
+
+* HTTP Error *401*(Unauthorized): When you try to access a restricted web page
+but aren't authorized to do so (usually because of a failed login attempt).
```
Comment (hellais) [inline content/about/glossary.md]: You should replace the above with: HTTP status xxx

File: content/about/glossary.md [hunk 21]
```diff
@@ -0,0 +1,975 @@
...
+A server normally responds with the content of the website it is hosting. This
+response is an HTTP response.
+
+### HTTP status codes
+
+HTTP status codes are standardized codes (maintained by the Internet Assigned
+Numbers Authority) that are issued by a server in response to a client's
+request. As suggested by their name, these codes communicate the status of an
+HTTP response.
+
```
Comment (hellais) [inline content/about/glossary.md]: I would add something saying:

HTTP status codes are divided into specific ranges, depending on the type of status they are communicating.

HTTP status codes that go from 200-299 are used to indicate a successful request
HTTP status codes that go from 300-399 are used to indicate that there are multiple choices, for example to redirect the request to another site
HTTP status codes that go from 400-499 are used to indicate errors in the request by the client
HTTP status code that go from 500-599 are used to indicate server side errors

File: content/about/glossary.md [hunk 22]
```diff
@@ -0,0 +1,975 @@
...
+Common HTTP status codes include:
+
```
Comment (hellais) [inline content/about/glossary.md]: We should also add:

* HTTP Status *200*(OK): Used to indicate that the request was successful. This is generally the status code used most commonly for successful page loads.

* HTTP Status *301*(Moved Permanently): When the resource requested by the client is now available at a different location.

* HTTP Status *302*(Found): When the resource requested by the client can be **temporarily accessed** at a different location.

Comment (hellais) [reply content/about/glossary.md]: Also maybe just for fun, we should add:
* HTTP Status 451 (Unavailable For Legal Reasons): When the resource that is being tried to access is blocked by the server due to some legal restrictions. This is basically the status code to be used for implementing internet censorship.

File: content/about/glossary.md (hunk 1 above)
Comment (agrabeli) [reply content/about/glossary.md]: Thanks, I've added a sentence to mention this.

File: content/about/glossary.md (hunk 2 above)
Comment (agrabeli) [reply content/about/glossary.md]: Thanks for the feedback guys. I've edited accordingly.

File: content/about/glossary.md (hunk 3 above)
Comment (agrabeli) [reply content/about/glossary.md]: Good catch! Edited.

File: content/about/glossary.md (hunk 4 above)
Comment (agrabeli) [reply content/about/glossary.md]: Thanks! This is very helpful.

File: content/about/glossary.md (hunk 5 above)
Comment (agrabeli) [reply content/about/glossary.md]: I think we referred to it as "cloud fronting" in our documentation, but I agree that it's probably best to include "domain fronting" in the glossary.

File: content/about/glossary.md (hunk 7 above)
Comment (agrabeli) [reply content/about/glossary.md]: +1

File: content/about/glossary.md (hunk 6 above)
Comment (agrabeli) [reply content/about/glossary.md]: This is great, thank you!

File: content/about/glossary.md (hunk 8 above)
Comment (agrabeli) [reply content/about/glossary.md]: OK

File: content/about/glossary.md (hunk 9 above)
Comment (agrabeli) [reply content/about/glossary.md]: I've edited this section per your suggestions.

File: content/about/glossary.md (hunk 10 above)
Comment (agrabeli) [reply content/about/glossary.md]: Thanks! Added.

File: content/about/glossary.md (hunk 11 above)
Comment (agrabeli) [reply content/about/glossary.md]: OK, I've added a sentence to link to the Web Connectivity test description.

File: content/about/glossary.md (hunk 12 above)
Comment (agrabeli) [reply content/about/glossary.md]: Thanks, edited.

File: content/about/glossary.md (hunk 13 above)
Comment (agrabeli) [reply content/about/glossary.md]: OK

File: content/about/glossary.md (hunk 14 above)
Comment (agrabeli) [reply content/about/glossary.md]: OK

File: content/about/glossary.md (hunk 15 above)
Comment (agrabeli) [reply content/about/glossary.md]: OK

File: content/about/glossary.md (hunk 16 above)
Comment (agrabeli) [reply content/about/glossary.md]: Thanks!

File: content/about/glossary.md (hunk 17 above)
Comment (agrabeli) [reply content/about/glossary.md]: OK. Is the remaining description of "HTTP request" accurate?

File: content/about/glossary.md (hunk 18 above)
Comment (agrabeli) [reply content/about/glossary.md]: OK

File: content/about/glossary.md (hunk 19 above)
Comment (agrabeli) [reply content/about/glossary.md]: Thanks!

File: content/about/glossary.md (hunk 20 above)
Comment (agrabeli) [reply content/about/glossary.md]: Sure

File: content/about/glossary.md (hunk 21 above)
Comment (agrabeli) [reply content/about/glossary.md]: Thanks, added!

File: content/about/glossary.md (hunk 22 above)
Comment (agrabeli) [reply content/about/glossary.md]: Thanks, I've added them all.

File: content/about/glossary.md [hunk 23]
```diff
@@ -0,0 +1,975 @@
...
+### HTTP request
+
+In order to connect to a server hosting an internet service (such as a website
+or application), you need to request access to it through the HTTP protocol.
+This is what we call an HTTP request.
+
+Every time you connect to a website, your browser (the client) sends a request
+("HTTP request") through the HTTP protocol to the server which is hosting that
+website. A server normally responds ("HTTP response") with the content of the
+website it is hosting.
```
Comment (hellais) [inline content/about/glossary.md]: I suggest replacing:

> A server normally responds ("HTTP response") with the content of the website it is hosting.

with

> A server normally replies with a "HTTP response" which includes the content of the website it is hosting.

File: content/about/glossary.md (hunk 17 above)
Comment (hellais) [reply content/about/glossary.md]: Added a small edit below.

**Comment by agrabeli** [comment]
In my latest commit I edit the glossary based on feedback shared by @hellais & @p1g30n. I would appreciate further review to prepare the glossary for publication.

**Comment by agrabeli** [comment]
@hellais My latest commit includes edits based on the last rounds of feedback. If the glossary looks good to you, please merge it.
